arrow-array/src/array/run_array.rs

```diff
@@ -121,6 +122,27 @@ impl<R: RunEndIndexType> RunArray<R> {
     pub fn values(&self) -> &ArrayRef {
         &self.values
     }
+
+    /// Downcast this dictionary to a [`TypedRunArray`]
+    ///
+    /// ```
+    /// use arrow_array::{Array, ArrayAccessor, RunArray, StringArray, types::Int32Type};
+    ///
+    /// let orig = [Some("a"), Some("b"), None];
+    /// let run_array = RunArray::<Int32Type>::from_iter(orig);
+    /// let typed = run_array.downcast_ref::<StringArray>().unwrap();
+    /// assert_eq!(typed.value(0), "a");
+    /// assert_eq!(typed.value(1), "b");
+    /// assert!(typed.values().is_null(2));
+    /// ```
+    ///
+    pub fn downcast_ref<V: 'static>(&self) -> Option<TypedRunArray<'_, R, V>> {
```

Comment:
```suggestion
    pub fn downcast<V: 'static>(&self) -> Option<TypedRunArray<'_, R, V>> {
```

Or something so as not to overload `dyn Any`